From d95ae843845760aecbbc62a734c2b93b401b1834 Mon Sep 17 00:00:00 2001
Message-Id: <d95ae843845760aecbbc62a734c2b93b401b1834.1624798048.git.mschiffer@universe-factory.net>
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sun, 27 Jun 2021 14:45:46 +0200
Subject: [PATCH] config: allow disabling L2TP offload when fastd doesn't
 support it

Only attempting to enable the offloading should raise an error when it
is not supported.
---
 src/config.y | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/config.y b/src/config.y
index a107d7b9fda7..7a3ec6a32649 100644
--- a/src/config.y
+++ b/src/config.y
@@ -282,12 +282,14 @@ offload:	TOK_L2TP boolean {
 #ifdef WITH_OFFLOAD_L2TP
 			conf.offload_l2tp = $2;
 #else
+			if ($2) {
 # ifdef __linux__
-			fastd_config_error(&@$, state, "L2TP offload is not supported by this build of fastd");
+				fastd_config_error(&@$, state, "L2TP offload is not supported by this build of fastd");
 # else
-			fastd_config_error(&@$, state, "L2TP offload is not supported on this platform");
+				fastd_config_error(&@$, state, "L2TP offload is not supported on this platform");
 # endif
-			YYERROR;
+				YYERROR;
+			}
 #endif
 		}
 	;
-- 
2.32.0

